### PR TITLE
WT-12902 Remove unused temporaries from IGNORE_RET macros

### PR DESCRIPTION
**Pull request by marcbutler**
Explicitly casting to void is not sufficient to suppress the warning. Inverting the value and then casting to void is works for all current mongodb v4 toolchains.

### CODE REVIEW

File: src/include/misc.h [hunk 1]
```diff
@@ -39,24 +39,13 @@
...
-    do {                           \
-        const void *__ignored_ret; \
-        __ignored_ret = (call);    \
-        WT_UNUSED(__ignored_ret);  \
-    } while (0)
+
+/* 
+ * Explicitly suppress unused return value warning at function call sites. 
+ */
+#define WT_IGNORE_RET(call) ((void)!(call))
```
Comment (ajmorton) [inline src/include/misc.h:46]: Question: Why do we need to `!` the result before `(void)`ing it?

File: src/include/misc.h [hunk 2]
```diff
@@ -39,24 +39,13 @@
...
+#define WT_IGNORE_RET_BOOL(call) ((void)!(call))
```
Comment (ajmorton) [inline src/include/misc.h:47]: Can we merge these macros into a single `WT_IGNORE_RET`?
We only have 7 instances of `WT_IGNORE_RET_BOOL` and `WT_IGNORE_RET_PTR` in `src` so the diff will be small.

**Review by korteland — APPROVED** [review]
The change itself LGTM, but I have one question:

> Explicitly casting to void is not sufficient to suppress the warning

Would this not apply to `WT_UNUSED` as well? Or is the compiler complaint specific to function calls?

Regarding Andrew's ask about combining them: I'd be happy to see it but I understand if you feel it's out of scope.

File: src/include/misc.h (hunk 1 above)
Comment (ershov) [reply src/include/misc.h:48]: GCC has a bug where `(void)` doesn't cancel the "unused" warning for functions marked as "warn_unused_result" ([link](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509)).

Comment (marcbutler) [reply src/include/misc.h:48]: Yes and it keeps raising it's head [link](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425#c34).
I had spoken to Andrew out of band, and was going to comment here after I had a chance to validate my reasoning with a compiler expert.

**Comment by marcbutler** [comment]
> Would this not apply to `WT_UNUSED` as well? Or is the compiler complaint specific to function calls?

As I understand it no as it is different warning: unused variable / parameter.
(As discussed.)

**Comment by marcbutler** [comment]
Outside of review @ershov asserted that this approach could be applied to both this and the `WT_UNUSED` case as originally posed by @korteland:

> Would this not apply to WT_UNUSED

I responded that I do not think it can work as the logical inversion `!` work around can only be used with scalar values.
Will update code comments accordingly.

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks for making the changes!

**Pull request merged**